### PR TITLE
[BugFix] fix sort key of materialized view

### LINKED ISSUE #28766: Allow use any columns as sort key of materialized view
### Steps to reproduce the behavior (Required)

Materialized view allow user manually specify the sort key:
```sql
CREATE MATERIALIZED VIEW `mv1`
ORDER BY (
REFRESH ASYNC
AS SELECT `lo_quantity`, `lo_revenue`, avg(`lo_tax`) AS `avg`
FROM `lineorder`
WHERE `lo_linenumber` = 1
GROUP BY 1, 2;
```

But actually the `ORDER BY` could only be the prefix of  `SELECT LIST`. So it doesn't make any sense.

SR should support specify any columns as the sort key in materialized view.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #28766

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -410,31 +412,41 @@ private List<Column> genMaterializedViewColumns(CreateMaterializedViewStatement
...
-                        throw new SemanticException("Sort key(%s) doesn't exist.", keyCols.get(i));
-                    }
-                    throw new SemanticException("Sort key should be a ordered prefix of select cols.");
-                }
+            if (CollectionUtils.isEmpty(statement.getSortKeys())) {
+                return mvColumns;
+            }
 
-                Type keyColType = column.getType();
+            // Reorder the MV columns according to sort-key
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:426]: Do this only when statement.getSortKeys is not empty ?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 28 (96.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|27|28|96.43%|[415]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 28 (96.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|27|28|96.43%|[415]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 27 / 28 (96.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|27|28|96.43%|[415]|

**Review by LiShuMing — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**